clang/include/clang/3C/CastPlacement.h

```diff
@@ -14,11 +14,31 @@
 
 #include "clang/3C/ConstraintResolver.h"
 #include "clang/AST/RecursiveASTVisitor.h"
+#include "RewriteUtils.h"
+
+// Locates expressions which are children of explicit cast expressions after
+// ignoring any intermediate implicit expressions introduced in the clang AST.
+class CastLocatorVisitor : public RecursiveASTVisitor<CastLocatorVisitor> {
+public:
+  explicit CastLocatorVisitor(ASTContext *C) : Context(C) {}
+
+  bool VisitCastExpr(CastExpr *C);
+
+  bool exprHasCast(Expr *E) const;
+
+private:
+  ASTContext *Context;
+  std::set<Expr *> ExprsWithCast;
+
+  static Expr *ignoreCheckedCImplicit(Expr *E);
+};
 
 class CastPlacementVisitor : public RecursiveASTVisitor<CastPlacementVisitor> {
 public:
-  explicit CastPlacementVisitor(ASTContext *C, ProgramInfo &I, Rewriter &R)
-      : Context(C), Info(I), Writer(R), CR(Info, Context) {}
+  explicit CastPlacementVisitor
+    (ASTContext *C, ProgramInfo &I, Rewriter &R, CastLocatorVisitor &L)
```

Comment:
Seems a little strange for one visitor to take another as a parameter. Is what's happening here, effectively, that the set of `ExprsWithCast` from the locator will be used by the `CastPlacementVisitor` ?